src/persist-client/src/cache.rs

```diff
@@ -62,7 +62,8 @@ pub struct PersistClientCache {
     pubsub_sender: Arc<dyn PubSubSender>,
     _pubsub_receiver_task: JoinHandle<()>,
 
-    // WIP hack plumb this around instead
+    // WIP hack plumb this around instead (What does this mean? Why not have this be `std::sync::OnceLock<Arc<TxnRead>>`?)
+    // This is the `TxnRead` for the `clusterd` process.
```

Comment:
mz_persist_client cannot depend on things in mz_persist_txn. the hack here is that we already have a `PersistClientCache` in all the places where we render the persist-txn operator (i.e. compute worker state), so I just reused that plumbing but probably better to re-plumb just the persist-txn specific stuff

> `Arc<TxnRead>`
also, probably want booting the TxnRead worker to be lazy, so we don't start one up on clusters that never read a table